File: D_star Lite/D_star Lite/CalculateKey.m
function [PrimaryKey,SecondaryKey]=CalculateKey(s)%计算节点key值,参数（当前节点）,返回该节点key值
    global k_m;
    global g;
    global rhs;
    global s_start;

    h=norm(s_start-s(1:2));
    SecondaryKey=min(   g(s(1),s(2)),     rhs(s(1),s(2))  );
    PrimaryKey=SecondaryKey+0.99*h+k_m;
end

File: d_star_lite_cpp/src/CalculateKey.cpp
/* Include files */
#include "../include/main.h"
#include "../include/Initialize.h"
#include "../include/CalculateKey.h"

/* Parameters Declarations */
static double SecondaryKey;
static double PrimaryKey;
static double h;

/* Matrix Declarations */
static RowVector2d key;

/* Function Declarations */
RowVector2d CalculateKey(RowVectorXi s)
{
    h = (s_start - s.head(2)).cast<double>().norm();
    SecondaryKey = fmin(g(s(0), s(1)), rhs(s(0), s(1)));
    PrimaryKey = SecondaryKey + 0.99 * h + k_m;
    key << PrimaryKey, SecondaryKey;
    return key;
}